internal/engine/handler.go

```diff
@@ -79,9 +93,24 @@ func (e *ExecutorEventHandler) Wait() {
 // HandleEntityEvent handles events coming from webhooks/signals
 // as well as the init event.
 func (e *ExecutorEventHandler) HandleEntityEvent(msg *message.Message) error {
-	// Grab the context before making a copy of the message
-	msgCtx := msg.Context()
-	// Let's not share memory with the caller
+
+	// NOTE: we're _deliberately_ "escaping" from the parent context's Cancel/Done
+	// completion, because the default watermill behavior for both Go channels and
+	// SQL is to process messages sequentially, but we need additional parallelism
+	// beyond that.  When we switch to a different message processing system, we
+	// should aim to remove this goroutine altogether and have the messaging system
+	// provide the parallelism.
+	// We _do_ still want to cancel on shutdown, however.
+	// TODO: Make this timeout configurable
+	msgCtx := context.WithoutCancel(msg.Context())
+	msgCtx, shutdownCancel := context.WithCancel(msgCtx)
+	{
+		e.lock.Lock()
+		defer e.lock.Unlock()
+		e.cancels = append(e.cancels, &shutdownCancel)
+	}
```

Comment:
**praise:** TIL that you can introduce arbitrary lexical scopes in Go as well.

Reply:
Same, when I saw it I actually ran a small code snip to make sure I was understanding :)

Reply:
Ha, I avoided screwing this up by accident!  It turns out that the other lock interaction was inside the goroutine, so this one sort of speeds through and doesn't run into the other lock anyway.  But I switched things up to reduce the scope of the lock by explicitly calling unlock on the following line.